apps/ios/Shared/Views/UserSettings/AdvancedNetworkSettings.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  AdvancedNetworkSettings.swift
+//  SimpleX (iOS)
+//
+//  Created by Evgeny on 02/08/2022.
+//  Copyright © 2022 SimpleX Chat. All rights reserved.
+//
+
+import SwiftUI
+import SimpleXChat
+
+private let secondsLabel =  NSLocalizedString("sec", comment: "network option")
+
+struct AdvancedNetworkSettings: View {
+    @State private var netCfg = NetCfg.defaults
+    @State private var currentNetCfg = NetCfg.defaults
+    @State private var cfgLoaded = false
+    @State private var enableKeepAlive = true
+    @State private var keepAliveOpts = KeepAliveOpts.defaults
+    @State private var showSaveErrorAlert = false
+    @State private var saveConfigError: String?
+
+    var body: some View {
+        VStack {
+            List {
+                Section {
+                    Button {
+                        currentNetCfg = NetCfg.defaults
+                        resetNetCfg()
+                        saveNetCfg()
+                    } label: {
+                        Text("Reset to defaults")
+                    }
+                    .disabled(currentNetCfg == NetCfg.defaults)
+
+                    timeoutSettingPicker("TCP connection timeout", selection: $netCfg.tcpConnectTimeout, values: [2_500000, 5_000000, 7_500000, 10_000000, 15_000000, 20_000000], label: secondsLabel)
+                    timeoutSettingPicker("Protocol timeout", selection: $netCfg.tcpTimeout, values: [1_500000, 3_000000, 5_000000, 7_000000, 10_000000, 15_000000], label: secondsLabel)
+                    timeoutSettingPicker("PING interval", selection: $netCfg.smpPingInterval, values: [120_000000, 300_000000, 600_000000, 1200_000000, 2400_000000], label: secondsLabel)
```

Comment:
maybe we could do these conversions somewhere in api (or even in backend but I'm not offering to move it there now).. just seems strange that top level ui needs to know about low level implementation scales

Reply:
I think at least we could make a local constant / a function to use in these lists, e.g. `toMicrosec = 1_000_000`? I just feel it's really easy to make an error here

Reply:
then the numbers would have to be doubles. I tried map, tried different things, this seems the least bad...